pkg/inspector/image-inspector.go

```diff
@@ -182,6 +179,91 @@ func (i *defaultImageInspector) Inspect() error {
 	return nil
 }
 
+func aggregateBytesAndReport(bytesChan chan int) {
+	var bytesDownloaded int = 0
+	timeOutChannel := time.After(PULL_LOG_INTERVAL_SEC * time.Second)
+	for {
+		select {
+		case bytes, open := <-bytesChan:
+			if !open {
+				return
+			}
+			bytesDownloaded += bytes
+		case <-timeOutChannel:
+			log.Printf("Pulling... %d KB", bytesDownloaded/1024)
+			timeOutChannel = time.After(PULL_LOG_INTERVAL_SEC * time.Second)
+		}
+	}
+}
+
+func decodeDockerPullMessages(bytesChan chan int, reader io.Reader) {
+	type progressDetailType struct {
+		Current, Total int
+	}
+	type pullMessage struct {
+		Status, Id     string
+		ProgressDetail progressDetailType
+	}
+	defer func() { close(bytesChan) }()           // Closing the channel to end the other routine
+	layersByetsDownloaded := make(map[string]int) // bytes downloaded per layer
+	dec := json.NewDecoder(reader)                // decoder for the json messages
+	for {
+		var v pullMessage
+		if err := dec.Decode(&v); err != nil {
+			if err != io.ErrClosedPipe {
+				log.Printf("Error decoding json: %v", err)
+			}
+			break
+		}
+		// decoding
+		if v.Status == "Downloading" {
+			bytes := v.ProgressDetail.Current
+			last, existed := layersByetsDownloaded[v.Id]
+			if !existed {
+				last = 0
+			}
+			layersByetsDownloaded[v.Id] = bytes
+			bytesChan <- (bytes - last)
+		}
+	}
+}
+
+// pullImage pulls the inspected image using the given client.
+// It will try to use all the given authentication methods and will fail
+// only if all of them failed.
+func (i *defaultImageInspector) pullImage(client *docker.Client) error {
+	log.Printf("Pulling image %s", i.opts.Image)
+
+	var imagePullAuths *docker.AuthConfigurations
+	var authCfgErr error
+	if imagePullAuths, authCfgErr = i.getAuthConfigs(); authCfgErr != nil {
+		return authCfgErr
+	}
+
+	reader, writer := io.Pipe()
+	// handle closing the reader/writer in the method that creates them
+	defer writer.Close()
+	defer reader.Close()
+	imagePullOption := docker.PullImageOptions{
+		Repository:    i.opts.Image,
+		OutputStream:  writer,
+		RawJSONStream: true,
+	}
+
+	bytesChan := make(chan int)
+	go aggregateBytesAndReport(bytesChan)
+	go decodeDockerPullMessages(bytesChan, reader)
+
+	// Try all the possible auth's from the config file
+	var authErr error
+	for _, auth := range imagePullAuths.Configs {
```

Comment:
I'm curious about how this will display and work if there are non-auth related errors during the pull attempts.  Reading the code I would expect something like:

```
Pulling... 10 kb
Pulling... 20 kb
Pulling... 30 kb
<unseen 500 error>
Pulling... 10 kb
Pulling... 20 kb
```
We might want to display the error.